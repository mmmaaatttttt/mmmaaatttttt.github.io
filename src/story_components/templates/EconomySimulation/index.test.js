import React from "react";
import { fireEvent, render } from "@testing-library/react";
import { EconomySimulation } from ".";

describe("smoke and snapshot tests", () => {
  it("renders successfully", () => {
    render(<EconomySimulation />);
  });

  it("renders first demonstration successfully", () => {
    const { asFragment } = render(<EconomySimulation idx={0} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders second demonstration successfully", () => {
    const { asFragment } = render(<EconomySimulation idx={1} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders first demonstration successfully", () => {
    const { asFragment } = render(<EconomySimulation idx={2} />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe("state changes", () => {
  it("shows 3 buttons once the simulation has started", () => {
    const { queryByText } = render(<EconomySimulation idx={0} />);

    const startBtn = queryByText("Start");
    expect(startBtn).toBeInTheDocument();
    fireEvent.click(startBtn);

    const chartBtn = queryByText("Show Chart");
    const resetBtn = queryByText("Reset");
    expect(startBtn).toHaveTextContent("Pause");
    expect(chartBtn).toBeInTheDocument();
    expect(resetBtn).toBeInTheDocument();
  });

  it("shows the bar graph when showingSimulation is false", () => {
    const { queryByText } = render(<EconomySimulation idx={0} />);

    const startBtn = queryByText("Start");
    fireEvent.click(startBtn);

    let text1 = queryByText("1");
    let text2 = queryByText("2");

    expect(text1).not.toBeInTheDocument();
    expect(text2).not.toBeInTheDocument();

    const chartBtn = queryByText("Show Chart");
    fireEvent.click(chartBtn);

    text1 = queryByText("1");
    text2 = queryByText("2");
    const nodesBtn = queryByText("Show Nodes");

    expect(text1).toBeInTheDocument();
    expect(text2).toBeInTheDocument();
    expect(nodesBtn).toBeInTheDocument();
  });

  it("removes the bar graph on reset", () => {
    const { queryByText } = render(<EconomySimulation idx={0} />);

    const startBtn = queryByText("Start");
    fireEvent.click(startBtn);

    const chartBtn = queryByText("Show Chart");
    fireEvent.click(chartBtn);

    let text1 = queryByText("1");
    let text2 = queryByText("2");

    const resetBtn = queryByText("Reset");
    fireEvent.click(resetBtn);

    expect(text1).not.toBeInTheDocument();
    expect(text2).not.toBeInTheDocument();
  });
});
