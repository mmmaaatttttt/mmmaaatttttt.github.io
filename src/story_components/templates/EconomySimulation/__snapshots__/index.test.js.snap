// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests renders first demonstration successfully 1`] = `
<DocumentFragment>
  <div
    class="sc-VigVT hSnLDN"
    width="50%"
  >
    <div
      class="sc-iwsKbI kvOfHz"
      width="auto"
    >
      <div
        class="sc-dVhcbM ecVKIK"
      >
        <div
          class="sc-bRBYWo gtDXeL"
        >
          <p
            class="sc-Rmtcm bImzes"
          >
            Population Size
          </p>
          <section>
            <p
              class="sc-hzDkRC kxpnkL"
            >
              <i
                class="sc-gZMcBi dPNFn fas fa-user fa-1x false"
                disabled=""
                style="color: rgb(160, 94, 82); opacity: 1;"
              />
            </p>
            <div
              class="rc-slider sc-chPdSV gcsTPk"
            >
              <div
                class="rc-slider-rail"
              />
              <div
                class="rc-slider-track"
                style="left: 0%; width: 0%;"
              />
              <div
                class="rc-slider-step"
              />
              <div
                aria-disabled="false"
                aria-valuemax="30"
                aria-valuemin="2"
                aria-valuenow="2"
                class="rc-slider-handle"
                role="slider"
                style="left: 0%; transform: translateX(-50%);"
                tabindex="0"
              />
              <div
                class="rc-slider-mark"
              />
              <div
                class="sc-iwsKbI gQUqIM"
                width="auto"
              >
                <div
                  class="sc-jAaTju gdaaGZ"
                  color="#c79991"
                  offset="10"
                  width="6"
                />
                <div
                  class="sc-jAaTju kbZoje"
                  color="#e9e9e9"
                  offset="10"
                  width="6"
                />
              </div>
            </div>
            <p
              class="sc-hzDkRC kxpnkL"
            >
              <i
                class="sc-gZMcBi dPNFn fas fa-users fa-1x false"
                disabled=""
                style="color: rgb(160, 94, 82); opacity: 1;"
              />
            </p>
          </section>
        </div>
      </div>
    </div>
    <div
      class="sc-iwsKbI QHnwj"
      width="auto"
    >
      <button
        class="sc-bxivhb cEXLum"
        color="#A05E52"
      >
        Start
      </button>
    </div>
    <div
      style="display: block;"
    >
      <div>
        <svg
          class="sc-gzVnrw jUuqtv"
          viewBox="0 0 600 600"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <clippath
              id="clip-path-simulation-0"
            >
              <rect
                height="600"
                width="600"
                x="0"
                y="0"
              />
            </clippath>
          </defs>
          <g
            clip-path="url(#clip-path-simulation-0)"
          >
            <g>
              <rect
                fill="none"
                height="600"
                stroke="#BBBBBB"
                stroke-width="3"
                width="600"
                x="0"
                y="0"
              />
              <g />
            </g>
          </g>
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`smoke and snapshot tests renders first demonstration successfully 2`] = `
<DocumentFragment>
  <div
    class="sc-VigVT hSnLDN"
    width="50%"
  >
    <div
      class="sc-iwsKbI kvOfHz"
      width="auto"
    >
      <div
        class="sc-dVhcbM ecVKIK"
      >
        <div
          class="sc-bRBYWo gtDXeL"
        >
          <p
            class="sc-Rmtcm bImzes"
          >
            Population Size
          </p>
          <section>
            <p
              class="sc-hzDkRC kxpnkL"
            >
              <i
                class="sc-gZMcBi dPNFn fas fa-user fa-1x false"
                disabled=""
                style="color: rgb(160, 94, 82); opacity: 1;"
              />
            </p>
            <div
              class="rc-slider sc-chPdSV gcsTPk"
            >
              <div
                class="rc-slider-rail"
              />
              <div
                class="rc-slider-track"
                style="left: 0%; width: 0%;"
              />
              <div
                class="rc-slider-step"
              />
              <div
                aria-disabled="false"
                aria-valuemax="30"
                aria-valuemin="2"
                aria-valuenow="2"
                class="rc-slider-handle"
                role="slider"
                style="left: 0%; transform: translateX(-50%);"
                tabindex="0"
              />
              <div
                class="rc-slider-mark"
              />
              <div
                class="sc-iwsKbI gQUqIM"
                width="auto"
              >
                <div
                  class="sc-jAaTju gdaaGZ"
                  color="#c79991"
                  offset="10"
                  width="6"
                />
                <div
                  class="sc-jAaTju kbZoje"
                  color="#e9e9e9"
                  offset="10"
                  width="6"
                />
              </div>
            </div>
            <p
              class="sc-hzDkRC kxpnkL"
            >
              <i
                class="sc-gZMcBi dPNFn fas fa-users fa-1x false"
                disabled=""
                style="color: rgb(160, 94, 82); opacity: 1;"
              />
            </p>
          </section>
        </div>
      </div>
    </div>
    <div
      class="sc-iwsKbI QHnwj"
      width="auto"
    >
      <button
        class="sc-bxivhb cEXLum"
        color="#A05E52"
      >
        Start
      </button>
    </div>
    <div
      style="display: block;"
    >
      <div>
        <svg
          class="sc-gzVnrw jUuqtv"
          viewBox="0 0 600 600"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <clippath
              id="clip-path-simulation-2"
            >
              <rect
                height="600"
                width="600"
                x="0"
                y="0"
              />
            </clippath>
          </defs>
          <g
            clip-path="url(#clip-path-simulation-2)"
          >
            <g>
              <rect
                fill="none"
                height="600"
                stroke="#BBBBBB"
                stroke-width="3"
                width="600"
                x="0"
                y="0"
              />
              <g />
            </g>
          </g>
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`smoke and snapshot tests renders second demonstration successfully 1`] = `
<DocumentFragment>
  <div
    class="sc-VigVT hSnLDN"
    width="50%"
  >
    <div
      class="sc-iwsKbI kvOfHz"
      width="auto"
    >
      <div
        class="sc-dVhcbM ecVKIK"
      >
        <div
          class="sc-bRBYWo gtDXeL"
        >
          <p
            class="sc-Rmtcm bImzes"
          >
            Population Size
          </p>
          <section>
            <p
              class="sc-hzDkRC kxpnkL"
            >
              <i
                class="sc-gZMcBi dPNFn fas fa-user fa-1x false"
                disabled=""
                style="color: rgb(160, 94, 82); opacity: 1;"
              />
            </p>
            <div
              class="rc-slider sc-chPdSV gcsTPk"
            >
              <div
                class="rc-slider-rail"
              />
              <div
                class="rc-slider-track"
                style="left: 0%; width: 0%;"
              />
              <div
                class="rc-slider-step"
              />
              <div
                aria-disabled="false"
                aria-valuemax="30"
                aria-valuemin="2"
                aria-valuenow="2"
                class="rc-slider-handle"
                role="slider"
                style="left: 0%; transform: translateX(-50%);"
                tabindex="0"
              />
              <div
                class="rc-slider-mark"
              />
              <div
                class="sc-iwsKbI gQUqIM"
                width="auto"
              >
                <div
                  class="sc-jAaTju gdaaGZ"
                  color="#c79991"
                  offset="10"
                  width="6"
                />
                <div
                  class="sc-jAaTju kbZoje"
                  color="#e9e9e9"
                  offset="10"
                  width="6"
                />
              </div>
            </div>
            <p
              class="sc-hzDkRC kxpnkL"
            >
              <i
                class="sc-gZMcBi dPNFn fas fa-users fa-1x false"
                disabled=""
                style="color: rgb(160, 94, 82); opacity: 1;"
              />
            </p>
          </section>
        </div>
      </div>
    </div>
    <div
      class="sc-iwsKbI QHnwj"
      width="auto"
    >
      <button
        class="sc-bxivhb cEXLum"
        color="#A05E52"
      >
        Start
      </button>
    </div>
    <div
      style="display: block;"
    >
      <div>
        <svg
          class="sc-gzVnrw jUuqtv"
          viewBox="0 0 600 600"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs>
            <clippath
              id="clip-path-simulation-1"
            >
              <rect
                height="600"
                width="600"
                x="0"
                y="0"
              />
            </clippath>
          </defs>
          <g
            clip-path="url(#clip-path-simulation-1)"
          >
            <g>
              <rect
                fill="none"
                height="600"
                stroke="#BBBBBB"
                stroke-width="3"
                width="600"
                x="0"
                y="0"
              />
              <g />
            </g>
          </g>
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
